fdbserver/include/fdbserver/TenantEntryCache.actor.h

```diff
@@ -342,25 +342,35 @@ class TenantEntryCache : public ReferenceCounted<TenantEntryCache<T>>, NonCopyab
 
 	void put(const TenantNameEntryPair& pair) {
 		TenantEntryCachePayload<T> payload = createPayloadFunc(pair.first, pair.second);
-		if (mapByTenantId.find(pair.second.id) == mapByTenantId.end()) {
-			ASSERT(mapByTenantName.find(pair.first) == mapByTenantName.end());
-
-			mapByTenantId[pair.second.id] = payload;
-			mapByTenantName[pair.first] = payload;
-
-			TraceEvent(SevInfo, "TenantEntryCacheNewTenant", uid)
-			    .detail("TenantName", pair.first.contents().toString())
-			    .detail("TenantID", pair.second.id)
-			    .detail("TenantPrefix", pair.second.prefix);
-		} else {
-			mapByTenantId[pair.second.id] = payload;
-			mapByTenantName[pair.first] = payload;
-
-			TraceEvent(SevInfo, "TenantEntryCacheUpdateTenant", uid)
-			    .detail("TenantName", pair.first.contents().toString())
-			    .detail("TenantID", pair.second.id)
-			    .detail("TenantPrefix", pair.second.prefix);
+		auto idItr = mapByTenantId.find(pair.second.id);
+		auto nameItr = mapByTenantName.find(pair.first);
+
+		Optional<TenantName> existingName;
+		Optional<int64_t> existingId;
+		if (nameItr != mapByTenantName.end()) {
+			existingId = nameItr->value.entry.id;
+			mapByTenantId.erase(nameItr->value.entry.id);
+		}
+		if (idItr != mapByTenantId.end()) {
+			existingName = idItr->value.name;
+			mapByTenantName.erase(idItr->value.name);
 		}
+
+		mapByTenantId[pair.second.id] = payload;
+		mapByTenantName[pair.first] = payload;
+
+		TraceEvent(SevInfo, "TenantEntryCachePut")
+		    .detail("TenantName", pair.first.contents().toString())
+		    .detail("TenantNameExisting", existingName.present() ? existingName.get().contents().toString() : "")
```

Comment:
Converting these types to a string isn't necessary, and it's actually undesirable because it causes the string to be double escaped. Instead, you can just log the string ref and optional types directly. Also, `SevInfo` isn't typically used either as it is the default:

```
TraceEvent("TenantEntryCachePut")
 		    .detail("TenantName", pair.first)
 		    .detail("TenantNameExisting", existingName)
 		    .detail("TenantID", pair.second.id)
 		    .detail("TenantIDExisting", existingId)
 		    .detail("TenantPrefix", pair.second.prefix);

Reply:
Great! Good to know that I can log optional directly.
Thanks for pointing out.